spec/board_spec.rb

```diff
@@ -0,0 +1,211 @@
+    board.mark_board(8, 'O')
+
+    expected_output  = " O |   |  \n"
+    expected_output += "-----------\n"
+    expected_output += "   | O |  \n"
+    expected_output += "-----------\n"
+    expected_output += "   |   | O\n"
+    expect(board.display).to eq(expected_output)
+  end
+
+  it 'returns a full board with arbitrary X and O values' do
+
+    board = Board.new
+
+    board.mark_board(0, 'O')
+    board.mark_board(1)
+    board.mark_board(2, 'O')
+    board.mark_board(3)
+    board.mark_board(4, 'O')
+    board.mark_board(5)
+    board.mark_board(6)
+    board.mark_board(7)
+    board.mark_board(8, 'O')
+
+    expected_output  = " O | X | O\n"
+    expected_output += "-----------\n"
+    expected_output += " X | O | X\n"
+    expected_output += "-----------\n"
+    expected_output += " X | X | O\n"
+    expect(board.display).to eq(expected_output)
+  end
+
+  it 'returns a full board with arbitrary character values' do
+
+    board = Board.new
+    board.mark_board(0, 'M')
+    board.mark_board(1, '$')
+    board.mark_board(2, 'O')
+    board.mark_board(3, '#')
+    board.mark_board(4, 'O')
+    board.mark_board(5)
+    board.mark_board(6)
+    board.mark_board(7, 'V')
+    board.mark_board(8, 'P')
+
+    expected_output  = " M | $ | O\n"
+    expected_output += "-----------\n"
+    expected_output += " # | O | X\n"
+    expected_output += "-----------\n"
+    expected_output += " X | V | P\n"
+    expect(board.display).to eq(expected_output)
+  end
+
+  it 'returns a full board with X values' do
+    board = Board.new
+
+    board.mark_board(0)
+    board.mark_board(1)
+    board.mark_board(2)
+    board.mark_board(3)
+    board.mark_board(4)
+    board.mark_board(5)
+    board.mark_board(6)
+    board.mark_board(7)
+    board.mark_board(8)
+
+    expected_output  = " X | X | X\n"
+    expected_output += "-----------\n"
+    expected_output += " X | X | X\n"
+    expected_output += "-----------\n"
+    expected_output += " X | X | X\n"
+    expect(board.display).to eq(expected_output)
+  end
+
+  it 'returns a full board with O values' do
+    board = Board.new
+
+    board.mark_board(0, 'O')
+    board.mark_board(1, 'O')
+    board.mark_board(2, 'O')
+    board.mark_board(3, 'O')
+    board.mark_board(4, 'O')
+    board.mark_board(5, 'O')
+    board.mark_board(6, 'O')
+    board.mark_board(7, 'O')
+    board.mark_board(8, 'O')
+
+    expected_output  = " O | O | O\n"
+    expected_output += "-----------\n"
+    expected_output += " O | O | O\n"
+    expected_output += "-----------\n"
+    expected_output += " O | O | O\n"
+    expect(board.display).to eq(expected_output)
+  end
+end
+
+describe '#position_taken?' do
+  it 'returns false when an index that is not taken is entered' do
+    board = Board.new
+
+    board.mark_board(6)
+    board.mark_board(7, 'O')
+    board.mark_board(8)
+
+    expect(board.position_taken?(4)).to eq(false)
+  end
+
+  it 'returns true when an index that is taken is entered' do
+
```

Comment:
🔍 Did you mean to put this empty line here? We don't usually put an empty line after `it`